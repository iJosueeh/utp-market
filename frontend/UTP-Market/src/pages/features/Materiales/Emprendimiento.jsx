import React, { useState } from "react";
import Tutoria from "../../../assets/img_M/tutoria.jpg";
import Guias from "../../../assets/img_M/guias.jpg";
import Ventas from "../../../assets/img_M/ventas.jpg";
import Material from "../../../assets/img_M/mt.jpg";
import "bootstrap/dist/css/bootstrap.min.css";
import { Modal, Button, Form, Card } from "react-bootstrap";

export default function Emprendimiento() {
  const [validated, setValidated] = useState(false);

  const handleSubmit = (event) => {
    const form = event.currentTarget;
    if (form.checkValidity() === false) {
      event.preventDefault();
      event.stopPropagation();
    }

    setValidated(true);
  };

  const [show, setShow] = useState(false);
  const [selected, setSelected] = useState(null);

  const handleClose = () => setShow(false);
  const handleShow = (item) => {
    setSelected(item);
    setShow(true);
  };

  const data = [
    {
      id: 1,
      titulo: "Tutorías",
      desc: "Apoyo académico entre estudiantes para reforzar materias clave.",
      img: Tutoria,
    },

    {
      id: 2,
      titulo: "Guías de estudio",
      desc: "Resúmenes y materiales hechos por estudiantes para facilitar el aprendizaje.",
      img: Guias,
    },
    {
      id: 3,
      titulo: "Ventas de snack",
      desc: "Snacks accesibles y prácticos dentro del campus.",
      img: Ventas,
    },
    {
      id: 4,
      titulo: "Materiales",
      desc: "Intercambio y venta de útiles y herramientas académicas.",
      img: Material,
    },
  ];

  return (
    <div className="container my-5">
      <h5 className="text-black text-center">MATERIALES</h5>
      <h2 className="text-center mb-4">Emprendimientos estudiantiles</h2>

      <div className="row">
        {data.map((item) => (
          <div className="col-md-3 mb-4" key={item.id}>
            <Card className="h-100 shadow-sm">
              <Card.Img variant="top"
                src={item.img}
                alt={item.titulo}
                className="img-fluid"
                style={{ height: "200px", objectFit: "cover" }}
              />
              <Card.Body>
                <Card.Title>{item.titulo}</Card.Title>
                <Card.Text>{item.desc}</Card.Text>
                <Button
                  variant="dark"
                  className="w-100"
                  onClick={() => handleShow(item)}
                ><p className="m-0">Más Información →</p>
                </Button>
              </Card.Body>
            </Card>
          </div>
        ))}
      </div>

      <Modal show={show} onHide={handleClose} centered>
        <Modal.Header closeButton>
          <Modal.Title>{selected?.titulo}</Modal.Title>
        </Modal.Header>
        <Modal.Body>
          <p>{selected?.desc}</p>
          <Form noValidate validated={validated} onSubmit={handleSubmit}>
            <Form.Group className="mb-3" controlId="validationCustom01">
              <Form.Label>Nombre</Form.Label>
              <Form.Control
                required
                type="text"
                placeholder="Ingresa tu nombre"
              />
              <Form.Control.Feedback type="invalid">
                Por favor, ingresa tu nombre.
              </Form.Control.Feedback>
            </Form.Group>
            <Form.Group className="mb-3" controlId="validationCustom02">
              <Form.Label>Correo</Form.Label>
              <Form.Control
                required
                type="email"
                placeholder="correo@ejemplo.com"
              />
              <Form.Control.Feedback type="invalid">
                Por favor, ingresa un correo válido.
              </Form.Control.Feedback>
            </Form.Group>
            <Form.Group className="mb-3" controlId="validationCustom03">
              <Form.Label>Mensaje</Form.Label>
              <Form.Control
                required
                as="textarea"
                rows={3}
                placeholder="Escribe tu mensaje aquí"
              />
              <Form.Control.Feedback type="invalid">
                Por favor, escribe un mensaje.
              </Form.Control.Feedback>
            </Form.Group>
          </Form>
        </Modal.Body>
        <Modal.Footer>
          <Button variant="secondary" onClick={handleClose}>
            Cerrar
          </Button>
          <Button variant="primary" onClick={handleSubmit}>
            Enviar
          </Button>
        </Modal.Footer>
      </Modal>
    </div>
  );
}
