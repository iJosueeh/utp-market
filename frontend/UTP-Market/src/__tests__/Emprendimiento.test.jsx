import React from 'react';
import { render, screen, fireEvent, within, waitFor } from "@testing-library/react";
import { describe, it, expect, vi } from "vitest";
import Emprendimiento from "../pages/features/Materiales/Emprendimiento";


vi.mock("../../../assets/img_M/tutoria.jpg", () => ({ default: "tutoria.jpg" }));
vi.mock("../../../assets/img_M/guias.jpg", () => ({ default: "guias.jpg" }));
vi.mock("../../../assets/img_M/ventas.jpg", () => ({ default: "ventas.jpg" }));
vi.mock("../../../assets/img_M/mt.jpg", () => ({ default: "mt.jpg" }));

describe("Emprendimiento component", () => {
  it("renderiza el título principal", () => {
    render(<Emprendimiento />);
    expect(screen.getByRole('heading', { name: /Emprendimientos estudiantiles/i, level: 2 })).toBeInTheDocument();
  });

  it("renderiza las cuatro tarjetas de emprendimiento", () => {
    const { container } = render(<Emprendimiento />);
    const cards = container.querySelectorAll('.card');
    expect(cards.length).toBe(4);

    const titles = ['Tutorías', 'Guías de estudio', 'Ventas de snack', 'Materiales'];
    titles.forEach(title => {
        expect(screen.getByText(title)).toBeInTheDocument();
    });
  });

  it("abre el modal con la información correcta al hacer clic en 'Más Información'", () => {
    render(<Emprendimiento />);
    
    const tutoriaCard = screen.getByText(/Tutorías/).closest('.card');
    const masInfoButton = within(tutoriaCard).getByRole('button', { name: /Más Información →/i });
    fireEvent.click(masInfoButton);

    const modal = screen.getByRole('dialog');
    expect(modal).toBeInTheDocument();
    expect(within(modal).getByText('Tutorías')).toBeInTheDocument();
  });

  it("cierra el modal al hacer clic en el botón de cierre del header", async () => {
    render(<Emprendimiento />);
    
    const masInfoButton = screen.getAllByRole('button', { name: /Más Información →/i })[0];
    fireEvent.click(masInfoButton);

    const modal = screen.getByRole('dialog');
    expect(modal).toBeInTheDocument();

    const closeButton = screen.getByLabelText('Close');
    fireEvent.click(closeButton);

    await waitFor(() => {
      expect(screen.queryByRole('dialog')).not.toBeInTheDocument();
    });
  });
});
