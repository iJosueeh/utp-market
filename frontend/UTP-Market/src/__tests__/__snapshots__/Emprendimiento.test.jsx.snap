// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Emprendimiento component > matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="container my-5"
  >
    <h5
      class="text-black text-center"
    >
      MATERIALES
    </h5>
    <h2
      class="text-center fw-bold mb-4"
    >
      Emprendimientos estudiantiles
    </h2>
    <div
      class="row"
    >
      <div
        class="col-md-3 mb-4"
      >
        <div
          class="card h-100 shadow-sm"
        >
          <img
            alt="Tutorías"
            class="card-img-top img-fluid"
            src="/src/assets/img_M/tutoria.jpg"
            style="height: 200px; object-fit: cover;"
          />
          <div
            class="card-body"
          >
            <h5
              class="card-title"
            >
              Tutorías
            </h5>
            <p
              class="card-text"
            >
              Apoyo académico entre estudiantes para reforzar materias clave.
            </p>
            <button
              class="btn btn-dark w-100"
              data-bs-target="#infoModal"
              data-bs-toggle="modal"
            >
              Más Información →
            </button>
          </div>
        </div>
      </div>
      <div
        class="col-md-3 mb-4"
      >
        <div
          class="card h-100 shadow-sm"
        >
          <img
            alt="Guías de estudio"
            class="card-img-top img-fluid"
            src="/src/assets/img_M/guias.jpg"
            style="height: 200px; object-fit: cover;"
          />
          <div
            class="card-body"
          >
            <h5
              class="card-title"
            >
              Guías de estudio
            </h5>
            <p
              class="card-text"
            >
              Resúmenes y materiales hechos por estudiantes para facilitar el aprendizaje.
            </p>
            <button
              class="btn btn-dark w-100"
              data-bs-target="#infoModal"
              data-bs-toggle="modal"
            >
              Más Información →
            </button>
          </div>
        </div>
      </div>
      <div
        class="col-md-3 mb-4"
      >
        <div
          class="card h-100 shadow-sm"
        >
          <img
            alt="Ventas de snack"
            class="card-img-top img-fluid"
            src="/src/assets/img_M/ventas.jpg"
            style="height: 200px; object-fit: cover;"
          />
          <div
            class="card-body"
          >
            <h5
              class="card-title"
            >
              Ventas de snack
            </h5>
            <p
              class="card-text"
            >
              Snacks accesibles y prácticos dentro del campus.
            </p>
            <button
              class="btn btn-dark w-100"
              data-bs-target="#infoModal"
              data-bs-toggle="modal"
            >
              Más Información →
            </button>
          </div>
        </div>
      </div>
      <div
        class="col-md-3 mb-4"
      >
        <div
          class="card h-100 shadow-sm"
        >
          <img
            alt="Materiales"
            class="card-img-top img-fluid"
            src="/src/assets/img_M/mt.jpg"
            style="height: 200px; object-fit: cover;"
          />
          <div
            class="card-body"
          >
            <h5
              class="card-title"
            >
              Materiales
            </h5>
            <p
              class="card-text"
            >
              Intercambio y venta de útiles y herramientas académicas.
            </p>
            <button
              class="btn btn-dark w-100"
              data-bs-target="#infoModal"
              data-bs-toggle="modal"
            >
              Más Información →
            </button>
          </div>
        </div>
      </div>
    </div>
    <div
      aria-hidden="true"
      aria-labelledby="infoModalLabel"
      class="modal fade"
      id="infoModal"
      tabindex="-1"
    >
      <div
        class="modal-dialog modal-dialog-centered"
      >
        <div
          class="modal-content"
        >
          <form
            class="needs-validation "
            novalidate=""
          >
            <div
              class="modal-header"
            >
              <h5
                class="modal-title"
                id="infoModalLabel"
              />
              <button
                aria-label="Cerrar"
                class="btn-close"
                data-bs-dismiss="modal"
                type="button"
              />
            </div>
            <div
              class="modal-body"
            >
              <p />
              <div
                class="mb-3"
              >
                <label
                  class="form-label"
                  for="nombre"
                >
                  Nombre
                </label>
                <input
                  class="form-control"
                  id="nombre"
                  required=""
                  type="text"
                />
                <div
                  class="invalid-feedback"
                >
                  Por favor, ingresa tu nombre.
                </div>
              </div>
              <div
                class="mb-3"
              >
                <label
                  class="form-label"
                  for="correo"
                >
                  Correo
                </label>
                <input
                  class="form-control"
                  id="correo"
                  required=""
                  type="email"
                />
                <div
                  class="invalid-feedback"
                >
                  Por favor, ingresa un correo válido.
                </div>
              </div>
              <div
                class="mb-3"
              >
                <label
                  class="form-label"
                  for="mensaje"
                >
                  Mensaje
                </label>
                <textarea
                  class="form-control"
                  id="mensaje"
                  required=""
                  rows="3"
                />
                <div
                  class="invalid-feedback"
                >
                  Por favor, escribe un mensaje.
                </div>
              </div>
            </div>
            <div
              class="modal-footer"
            >
              <button
                class="btn btn-secondary"
                data-bs-dismiss="modal"
                type="button"
              >
                Cerrar
              </button>
              <button
                class="btn btn-primary"
                type="submit"
              >
                Enviar
              </button>
            </div>
          </form>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
